### PR TITLE
Ui updates

### PR DESCRIPTION
**Pull request by Luminesce0**
**A commit for UI updates** 

**Fixes:**
- Copy URL FAB no longer crashes the app
- Fixed Spinner widget pushing up ImageView beyond
 
**Features:**
- Migration to androidx
- Gradle updates
- FABs/FAMs that align further align with Material Design Guidelines
- New preference setup
- Removal of unused ViewPager tabs

### CODE REVIEW

**Comment by Luminesce0** [comment]
I'm also going to fix the cardview size on the release commit/merge!

Have it extended for testing purposes.

File: app/src/main/java/net/wetfish/wetfish/ui/behaviors/FAMShrinkBehavior.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    @Override
+    public boolean layoutDependsOn(CoordinatorLayout parent, RelativeLayout child, View dependency) {
+        return dependency instanceof Snackbar.SnackbarLayout;
+    }
+
+    @Override
+    public boolean onDependentViewChanged(CoordinatorLayout parent, RelativeLayout child, View dependency) {
+        float translationY = Math.min(0, dependency.getTranslationY() - dependency.getHeight());
+        // Note that the RelativeLayout gets translated.
```
Comment (QuercusFelis) [inline app/src/main/java/net/wetfish/wetfish/ui/behaviors/FAMShrinkBehavior.java:43]: Might want more comments on this new code, but it's good you do have some.

File: app/src/main/java/net/wetfish/wetfish/ui/settings/MySettingsActivity.java [hunk 2]
```diff
@@ -0,0 +1,227 @@
+package net.wetfish.wetfish.ui.settings;
```
Comment (QuercusFelis) [inline app/src/main/java/net/wetfish/wetfish/ui/settings/MySettingsActivity.java:1]: Once again, more comments would be ideal.

**Review by QuercusFelis — APPROVED** [review]
Looks good for now before a new release though I'd like to see more comments to make interpreting this code easier.

File: app/src/main/java/net/wetfish/wetfish/ui/behaviors/FAMShrinkBehavior.java (hunk 1 above)
Comment (Luminesce0) [reply app/src/main/java/net/wetfish/wetfish/ui/behaviors/FAMShrinkBehavior.java:43]: I'll look it over and see if I can put some m ore comments on an upcoming commit. Thanks!

File: app/src/main/java/net/wetfish/wetfish/ui/settings/MySettingsActivity.java (hunk 2 above)
Comment (Luminesce0) [reply app/src/main/java/net/wetfish/wetfish/ui/settings/MySettingsActivity.java:1]: I'll look it over and see if I can put some m ore comments on an upcoming commit. Thanks!

**Pull request merged**